src/expression.js

```diff
@@ -592,16 +592,26 @@ pp.parseFunctionBody = function(node, allowExpression) {
   // are not repeated, and it does not try to bind the words `eval`
   // or `arguments`.
   if (this.strict || !isExpression && node.body.body.length && this.isUseStrict(node.body.body[0])) {
-    let nameHash = {}, oldStrict = this.strict
+    let oldStrict = this.strict
     this.strict = true
     if (node.id)
       this.checkLVal(node.id, true)
-    for (let i = 0; i < node.params.length; i++)
-      this.checkLVal(node.params[i], true, nameHash)
+    this.checkParams(node);
     this.strict = oldStrict
+  } else if (isArrowFunction) {
+    this.checkParams(node);
```

Comment:
I think separate argument + check here is not needed as you can just call `this.checkParams` right inside of `parseArrowExpression` after `parseFunctionBody`.